frontend/src/api/prometheus/__tests__/useQueryRangeResourceData.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { TimeframeTitle } from '~/pages/modelServing/screens/types';
+import useQueryRangeResourceData from '~/api/prometheus/useQueryRangeResourceData';
+import usePrometheusQueryRange from '~/api/prometheus/usePrometheusQueryRange';
+import useRestructureContextResourceData from '~/utilities/useRestructureContextResourceData';
+import { TimeframeStep, TimeframeTimeRange } from '~/pages/modelServing/screens/const';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { mockPrometheusQueryResponseData } from '~/__mocks__/mockPrometheusQueryResponseData';
+
+jest.mock('~/api/prometheus/usePrometheusQueryRange');
+jest.mock('~/utilities/useRestructureContextResourceData');
```

Comment:
It is unnecessary to mock `useRestructureContextResourceData` because it is an internal detail.

I know that I said we can mock out the functions being wrapped but that's because the main target of the wrapper is `usePrometheusQueryRange` and not anything else. Instead of mocking `usePrometheusQueryRange` you can also use a spy to assert that the inputs are passed down to `usePrometheusQueryRange` correctly and then use the actual return value from `usePrometheusQueryRange`. There would still be no need to test every possible input to `usePrometheusQueryRange` because we only need to test that, as a wrapping function, the inputs are passed down and the outputs are passed up.

The purpose of this wrapper is to return the exact same result as `usePrometheusQueryRange` but as an object and not an array. 

In most tests, less mocking is best.

Previously you were testing all permutations of inputs that got passed down to `usePrometheusQueryRange` which was overkill when testing a wrapper because that's what the tests for `usePrometheusQueryRange` are for.

Reply:
@christianvogt updated the PR accordingly. It is up for review. Thanks!